### DISCUSSION TITLE
How to display notification in foreground

### DISCUSSION THREAD

**Discussion by saurabh874** [invertase/react-native-firebase#4325] [Q&A] [answered]
@mikehardy i am using react-native-firebase/app -8.4.0
and react-native-firebase/message-7.8.3
in background and kill mode notification successfully displaying but in foreground notification is not displaying?
even its calling messaging().onMessage every time when new notification come up in foreground.
so there is any thing you can help me?
i do not want use any 3rd like notifee or react-native-push-notification
please help
can we use
**const localNotif = new firebase.notifications.Notification()
.setTitle('Gazelkin')
.setBody(message.data.message)
.setSound('default')

firebase.notifications().displayNotification(localNotif).catch(err => console.error(err))**
in onMessage ?

**Comment by TommyLeong** [comment]
I'm curious how do you show your notification banner in banner and kill state since you are not using any 3rd party lib. 

Well for Foreground, I'm having the following code at my `AppDelegate.m` which renders my notification when app is at foreground
```
-(void)userNotificationCenter:(UNUserNotificationCenter *)center willPresentNotification:(UNNotification *)notification withCompletionHandler:(void (^)(UNNotificationPresentationOptions options))completionHandler
{
  NSDictionary *userInfo = notification.request.content.userInfo;
  NSLog(@"%@", userInfo);
  completionHandler(UNAuthorizationOptionSound | UNAuthorizationOptionAlert | UNAuthorizationOptionBadge);
}

```

**Comment by mikehardy** [comment, marked as answer]
You will need to integrate a 3rd party notification library. I would embrace it vs rejecting it. Proper on-device notification handling is an important set of features for an app and should be treated as such.

Notifications can be displayed easily on device with the built in handling of notification payloads in all but foreground state @TommyLeong as @saurabh874 mentions. But the underlying SDKs don't handle notification payloads if the app is foreground. That's what a real notification package is for.

**Reply by TommyLeong** [reply]
Certainly, Mike is right. I do have 3rd party notification library installed. I'm using https://github.com/zo0r/react-native-push-notification and https://github.com/react-native-community/push-notification-ios to handle notifications.
